src/util/system.cpp

```diff
@@ -67,6 +67,31 @@
 
 #ifdef HAVE_SYS_PRCTL_H
 #include <sys/prctl.h>
+// Set spectre prctl defines so that we can build on older kernels
```

Comment:
This doesn't seem safe. What if a kernel has number 52 with a different meaning?

Reply:
AFAIK the linux kernel API is supposed to be stable for these types of calls so that should not change.

Reply:
Should make it clear that this is for *linux only*.
Are there other operating systems with PRCTL?

Reply:
I think some emulate the interface.

Reply:
IRIX has its own prctl(), and there's no guarantee someone won't fork Linux incompatibly either - it's open source, after all.

Anyone building from source should have recent enough headers to match their actual kernel anyway. Let's just make sure the gitian builds have it working, and rely on the system for source builds...?

Reply:
Irix is an abandoned platform, I think the only other non-linux OS's that have prctl are BSD's but they seem to be just for compatibility with linux so their prctl numbers should always match. Regardless as long as those values are present in the kernel header the ones I hard coded should not get used. I didn't want to have it build this feature based on the presence of the defines in prctl.h since that would complicate the testing and logic since some of the options may not be present. For example I think `PR_SPEC_FORCE_DISABLE` was added after `PR_SPEC_DISABLE`.

If linux were to change the prctl number for these it would break on binary builds regardless, but that sounds quite unlikely with the current linux userspace API stability policy.